services/src/modules/directives/rest/types.ts

```diff
@@ -1,6 +1,7 @@
 export interface KeyValue {
   key: string;
   value: string;
+  required?: boolean;
```

Comment:
This interface is in use for request headers too. So we need to implement the same functionality in `parseHeaders` function.

Reply:
Makes sense to also support the same logic for optionally requiring headers